pkg/eestream/encode.go

```diff
@@ -130,31 +165,32 @@ func (ep *encodedPiece) Read(p []byte) (n int, err error) {
 	copy(outbufs[i], outbufs[i][n:])
 	// and shrink the buffer
 	outbufs[i] = outbufs[i][:len(outbufs[i])-n]
-	// if there's nothing left, decrement the amount of pieces we have
-	if len(outbufs[i]) <= 0 {
-		ep.er.piecesRemaining--
-	}
 	return n, nil
 }
 
 // EncodedRanger will take an existing Ranger and provide a means to get
 // multiple Ranged sub-Readers. EncodedRanger does not match the normal Ranger
 // interface.
 type EncodedRanger struct {
-	es ErasureScheme
-	rr ranger.Ranger
+	es              ErasureScheme
+	rr              ranger.Ranger
+	maxBufferMemory int
 }
 
 // NewEncodedRanger creates an EncodedRanger
-func NewEncodedRanger(rr ranger.Ranger, es ErasureScheme) (*EncodedRanger,
-	error) {
+func NewEncodedRanger(rr ranger.Ranger, es ErasureScheme,
+	maxBufferMemory int) (*EncodedRanger, error) {
```

Comment:
should we just make this a uint ?

Reply:
downside of uints is there's weird behavior around 0, which is a pretty common value. less bugprone imo to use an int. people using uints often mess up on decrementing for loops, or casts of negative values, etc.

Reply:
I also noticed the issues JT is talking about and I don't see a common practice of using unit instead of int for variables that are semantically non-negative. So, I'll leave it as is for now.